docs/components/shell/TopNav.tsx

```diff
@@ -50,9 +54,27 @@ export const TopNav = ({
           </Link>
         </div>
 
-        <Link href={"https://github.com/glow-xyz/nftoken"}>
-          <a>GitHub</a>
-        </Link>
+        <div className="flex-center">
+          <Link href={"https://github.com/glow-xyz/nftoken"}>
+            <a>GitHub</a>
+          </Link>
+
+          {userContext && !userContext.user && userContext.hasGlow && (
+            <button onClick={userContext.signIn} className="auth-button">
+              Sign in with Glow
+            </button>
+          )}
+
+          {userContext && !userContext.user && !userContext.hasGlow && (
+            <a
```

Comment:
Let's use the `<GlowSignInButton />` that Pablo made here: https://github.com/luma-team/glow-js/tree/master/packages/glow-react

If that button has some issues, we can fix them. With that button we shouldn't need the `setTimeout` stuff.

Reply:
Would it make sense to swap out my `UserProvider` with Pablo’s `GlowProvider` entirely? Or should I keep the custom `UserProvider` and just use that button?

Reply:
Good idea!